src/intersections/lineraysegmentplane.jl

```diff
@@ -0,0 +1,111 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+function lineraysegmentdirection(x::Line)
+  return x(1) - x(0)
+end
+
+function lineraysegmentdirection(x::Ray)
+  return direction(x)
+end
+
+function lineraysegmentdirection(x::Segment)
+  return -reduce(-, coordinates.(vertices(x)))
+end
+
+function lineraysegmentorigin(x::Line)
+  return x(0)
+end
+
+function lineraysegmentorigin(x::Ray)
+  return origin(x)
+end
+
+function lineraysegmentorigin(x::Segment)
+  return vertices(x)[1]
+end
```

Comment:
Can these three methods be replaced by the generic x(0) that works for all three types?